sunpy/physics/differential_rotation.py

```diff
@@ -593,6 +593,12 @@ def differential_rotate(smap, observer=None, time=None, **diff_rot_kwargs):
     for key in all_keys:
         out_meta.pop(key)
 
+    if 'rsun_ref' not in out_meta:
+        warn_user(
+            f"'rsun_ref' not found in metadata. Using a default value {smap.rsun_meters} "
+            "If this value is incorrect, you will need to add the correct value to the map metadata."
+        )
+        out_meta['rsun_ref'] = smap.rsun_meters.to_value(u.m)
     # Add a new HGS observer
     out_meta.update(get_observer_meta(new_observer, out_meta['rsun_ref']*u.m))
```

Comment:
The straightforward code fix is to simply change `out_meta['rsun_ref']*u.m` to `smap.rsun_meters`.  This should naturally trigger the warning in `mapbase.py` when there needs to be an assumption for the radius of the Sun.
```suggestion
    # Add a new HGS observer
    out_meta.update(get_observer_meta(new_observer, smap.rsun_meters))
```

Reply:
You are right. This is giving default radius as well as it's warning is present in mapbase.py. I'll modify test cases for the same.